omniscidb/QueryEngine/QueryTemplateGenerator.cpp

```diff
@@ -561,6 +552,7 @@ class GroupByQueryTemplateGenerator : public QueryTemplateGenerator {
     auto crt_matched_uncasted_ptr =
         new llvm::AllocaInst(i32_type, 0, "crt_matched", bb_entry);
     if (crt_matched_uncasted_ptr->getType() != pi32_type) {
+      // can this ever happen?
```

Comment:
I think it always happens if recall correctly. Alloca returns a pointer type with the default address space, i.e. zero.